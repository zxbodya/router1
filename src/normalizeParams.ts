import { RouteParams } from './Router';

const has = Object.prototype.hasOwnProperty;

// values like would be parsed from generated url
export function normalizeParams(
  searchParams: string[],
  params: { [key: string]: any }
): RouteParams {
  const res = {} as RouteParams;
  for (const key in params) {
    /* istanbul ignore else */
    if (has.call(params, key)) {
      res[key] = `${params[key]}`;
    }
  }

  for (let i = 0, l = searchParams.length; i < l; i += 1) {
    const paramName = searchParams[i];
    if (paramName in params) {
      res[paramName] =
        typeof params[paramName] === 'boolean'
          ? params[paramName]
          : `${params[paramName]}`;
    }
  }
  return res;
}
