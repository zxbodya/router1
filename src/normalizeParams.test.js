import { normalizeParams } from './normalizeParams';

describe('normalizeParams', () => {
  it('converts path params to string', () => {
    expect(
      normalizeParams([], { a: 1, b: {}, c: false, d: true })
    ).toMatchSnapshot();
  });
  it('preserves boolean values in search params', () => {
    expect(
      normalizeParams(['c', 'd'], { a: 1, b: {}, c: false, d: true })
    ).toMatchSnapshot();
  });
});
