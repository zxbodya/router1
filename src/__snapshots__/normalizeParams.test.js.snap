// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`normalizeParams converts path params to string 1`] = `
Object {
  "a": "1",
  "b": "[object Object]",
  "c": "false",
  "d": "true",
}
`;

exports[`normalizeParams preserves boolean values in search params 1`] = `
Object {
  "a": "1",
  "b": "[object Object]",
  "c": false,
  "d": true,
}
`;
